lib/semantics/check-select-stmt.cc

```diff
@@ -0,0 +1,103 @@
+//= check-select-stmt.cc - Checker for select-case, select-rank, select-type ==
```

Comment:
The code below seems to only be checking `SELECT CASE`, not `SELECT RANK` or `SELECT TYPE`.  Aren't those two already done in name resolution?  If this code is going to check `SELECT RANK` or `SELECT TYPE`, please add a `TODO` comment to that effect; otherwise, don't mention them here.

Reply:
I, Sameeran and Anchu are working on SELECT CASE, SELECT RANK AND SELECT TYPE.
In my code i have added TODO for SELECT RANK AND SELECT TYPE.
We will update this as and when we add support for these constructs.